grpc-fixture/fixture/fixture_interceptor.go

```diff
@@ -52,15 +77,33 @@ func (f fixture) intercept(srv interface{}, ss grpc.ServerStream, info *grpc.Str
 			var receivedMessage []byte
 			if unprocessedReceivedMessage == nil {
 				err := ss.RecvMsg(&receivedMessage)
-				unprocessedReceivedMessage = receivedMessage
 				if err != nil {
 					return err
 				}
+				unprocessedReceivedMessage = receivedMessage
+
+				if info.FullMethod == "/s12.tasks.v1.ActionsService/GetAction" {
+					receivedMessageStructure := internal.Message{
+						MessageOrigin: internal.ClientMessage,
+						RawMessage:    receivedMessage,
+						Message:       nil,
+						Timestamp:     time.Time{},
+					}
+					receivedMessageDecoded, decodeErr := f.decoder.Decode(info.FullMethod, &receivedMessageStructure)
+					if receivedMessageDecoded == nil {
```

Comment:
would it be enough to just check `if decodeErr != nil ` ?

Reply:
Yep, was thinking to improve it as well